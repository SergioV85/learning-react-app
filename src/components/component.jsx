import React from 'react';

export class Subtitle extends React.Component {
  render() {
    return <h2>This is subtitle, simple component</h2>;
  }
}