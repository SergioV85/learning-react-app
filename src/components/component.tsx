import * as React from 'react';

export class Subtitle extends React.Component {
  public render() {
    return <h2 className="custom-header">This is subtitle, simple component</h2>;
  }
}