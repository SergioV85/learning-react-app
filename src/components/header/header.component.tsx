import * as React from 'react';

interface IHeaderProps {
  showBackButton?: boolean;
  onBackToList(): void;
}
// tslint:disable:max-line-length
export const Header = (props: IHeaderProps) =>
  <div className='c-header'>
    <span className='c-header__logo'>netflixroulette</span>
    {props.showBackButton ? <button className='btn c-header__back-button' onClick={props.onBackToList}> Search </button> : null}
  </div>;
// tslint:enable:max-line-length
