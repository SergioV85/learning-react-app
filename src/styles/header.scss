@import './variables';

.c-header {
  align-items: center;
  display: flex;
  justify-content: space-between;

  &__logo {
    color: $primary-red;
    font-family: $tilte-font;
    font-size: 1rem;
    font-weight: 700;
  }

  &__back-button {
    color: $primary-red;
    font-family: $main-font;
    font-size: 0.8rem;
  }
}