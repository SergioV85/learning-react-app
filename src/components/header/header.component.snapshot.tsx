import { shallow } from 'enzyme';
import * as React from 'react';
import { Header } from './header.component';

describe('Header Component Snapshot', () => {
  const mockedBackToList = jest.fn();

  it('should renders correctly', () => {
    const headerSnapshot = shallow(
      <Header showBackButton={true} onBackToList={mockedBackToList} />,
    );
    expect(headerSnapshot)
      .toMatchSnapshot();
  });
});
