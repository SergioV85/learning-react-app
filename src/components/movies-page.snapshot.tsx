import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { Header } from './header/header.component';

describe('Subtitle Snapshot', () => {
  const mockedBackToList = () => {};

  test('renders', () => {
    const component = renderer.create(
      <Header onBackToList={mockedBackToList} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});