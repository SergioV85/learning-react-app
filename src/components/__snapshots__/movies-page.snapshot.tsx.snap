// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movies Page Snapshot should render default page 1`] = `
<div
  className="c-page"
>
  <div
    className="c-page__header"
  >
    <div
      className="c-page__header-shadow"
    >
      <div
        className="p-2 container"
      >
        <Component
          onBackToList={[MockFunction]}
          showBackButton={false}
        />
        <ErrorBoundary
          errorMessage="Unfortunately, we can not show search panel"
          key="search-panel-wrapper"
        >
          <SearchPanel
            onChangeType={[MockFunction]}
            onInputChange={[MockFunction]}
            onSearchMovies={[MockFunction]}
            search=""
            searchType="title"
          />
        </ErrorBoundary>
      </div>
    </div>
  </div>
  <div
    className="c-page__status-bar"
  >
    <StatusBarPanel
      currentSortingOrder="asc"
      currentSortingType="rating"
      foundMovies={0}
      onChangeSortingOrder={[MockFunction]}
      onChangeSortingType={[MockFunction]}
    />
  </div>
  <div
    className="c-page__movies"
  >
    <ErrorBoundary
      errorMessage="Unfortunately, we cant show message"
      key="movies-empty-message-wrapper"
    >
      <Component />
    </ErrorBoundary>
  </div>
  <div
    className="c-page__footer"
  >
    <div
      className="p-2 container"
    >
      <Component />
    </div>
  </div>
</div>
`;
